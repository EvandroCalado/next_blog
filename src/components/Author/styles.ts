import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 3rem;
  margin: ${theme.spacings.xxlarge} 0;
`}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 15rem;
    border: 3px solid ${theme.colors.secondary};
    border-radius: 50%;
  `}
`;

export const Container = styled.div`
`;

export const Title = styled.h2`
${({ theme }) => css`
font-size: ${theme.font.sizes.xlarge};
`}
`;

export const Paragraph = styled.p`
`;
