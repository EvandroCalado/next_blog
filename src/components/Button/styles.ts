import styled, { css } from 'styled-components';
import { ButtonProps } from './';

export const Button = styled.button<Pick<ButtonProps, 'color'>>`
  ${({ theme, color }) => css`
    background-color: ${theme.colors[color]};
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.medium};
    border: none;
    padding: ${theme.spacings.xsmall} ${theme.spacings.medium};
    margin: ${theme.spacings.large} 0;
    border-radius: 0.5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all 300ms ease-in-out;
    cursor: pointer;

    &:hover {
      filter: brightness(80%);
    }

    &:disabled {
      background-color: ${theme.colors.mediumGray};
      color: ${theme.colors.darkGray};
      cursor: not-allowed;

      &:hover {
        filter: none;
      }
    }

    > svg {
      width: 3rem;
      margin-left: 1rem;
    }
  `}
`;
