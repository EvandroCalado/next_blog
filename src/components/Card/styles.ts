import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    a {
      transition: all 300ms ease-in-out;
    }

    a:hover {
      opacity: 1;
      filter: brightness(170%);
    }
`;

export const Cover = styled.div`
  ${({ theme }) => css`
    img {
      width: 100%;
      display: block;
      aspect-ratio: 16/9;
      border-radius: .5rem;
      margin-bottom: ${theme.spacings.small};

    }
  `}
`;

export const Heading = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.smallMedium};
    color: ${theme.colors.darkGray};
    margin-bottom: ${theme.spacings.xsmall};
  `}
`;

export const Paragraph = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    margin-bottom: ${theme.spacings.small};
    color: ${theme.colors.primary};
  `}
`;
