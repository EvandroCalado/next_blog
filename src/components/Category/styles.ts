import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 2rem;
  margin: ${theme.spacings.xxlarge} 0;
`}
`;

export const Image = styled.img`
  width: 8rem;
  border-radius: 0.3rem;
`;

export const Title = styled.h2`
${({ theme }) => css`
  font-size: ${theme.font.sizes.xlarge};

  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.large};
  }
`}
`;
