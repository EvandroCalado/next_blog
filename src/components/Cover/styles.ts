import styled, { css } from 'styled-components';

export const Image = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    aspect-ratio: 16/9;
    object-fit: cover;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
