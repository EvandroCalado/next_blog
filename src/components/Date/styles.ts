import styled, { css } from 'styled-components';

export const Date = styled.span`
  ${({ theme }) => css`
    color: ${theme.colors.secondary};
    font-style: italic;
    margin-right: 1rem;
  `}
`;
