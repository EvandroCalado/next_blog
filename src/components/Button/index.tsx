import React from 'react';
import * as Styled from './styles';

export type ButtonProps = {
  children: React.ReactNode;
  color: 'primary' | 'secondary';
  size: 'small' | 'medium' | 'large';
  icon?: React.ReactNode;
  disasbled?: boolean;
};

const Button = ({
  children,
  color,
  size,
  icon,
  disasbled = false,
}: ButtonProps) => {
  return (
    <Styled.Button disabled={disasbled} color={color} size={size}>
      {children}
      {icon && icon}
    </Styled.Button>
  );
};

export default Button;
