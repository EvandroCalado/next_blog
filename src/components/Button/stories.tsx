import Button, { ButtonProps } from './';
import { Meta, StoryFn } from '@storybook/react';

import { ChevronRight } from '@styled-icons/material-outlined';

export default {
  title: 'Button',
  component: Button,
  args: {
    children: 'Button',
  },
} as Meta;

export const WithIcon: StoryFn<ButtonProps> = (args) => (
  <div>
    <Button {...args} icon={<ChevronRight />} />
  </div>
);

export const WithSecondaryColor: StoryFn<ButtonProps> = (args) => (
  <div>
    <Button {...args} icon={<ChevronRight />} color="secondary" />
  </div>
);

export const WithoutIcon: StoryFn<ButtonProps> = (args) => (
  <div>
    <Button {...args} />
  </div>
);

export const Disabled: StoryFn<ButtonProps> = (args) => (
  <div>
    <Button {...args} disasbled={true} />
  </div>
);
