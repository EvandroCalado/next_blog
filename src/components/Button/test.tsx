import Button from './';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';

import { ChevronRight } from '@styled-icons/material-outlined';

describe('<Button />', () => {
  it('should render a button with default values', () => {
    renderTheme(
      <Button color="primary" size="medium">
        Button
      </Button>,
    );
    const button = screen.getByRole('button');

    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent('Button');
    expect(button).not.toBeDisabled();
    expect(screen.queryByRole('img')).not.toBeInTheDocument();
  });

  it('should render a button with values', () => {
    renderTheme(
      <Button
        color="primary"
        size="medium"
        icon={<ChevronRight />}
        disasbled={true}
      >
        Button
      </Button>,
    );
    const button = screen.getByRole('button');

    expect(button).toHaveAttribute('color', 'primary');
    expect(button).toHaveStyle({ 'font-size': '2.4rem' });
    expect(button.lastChild).toBeInTheDocument;
    expect(button).toBeDisabled();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <Button color="primary" size="medium">
        Button
      </Button>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
