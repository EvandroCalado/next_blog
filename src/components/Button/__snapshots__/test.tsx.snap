// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
.c0 {
  background-color: #252422;
  color: #f2f0d8;
  font-weight: 200;
  font-size: 2.4rem;
  border: none;
  padding: 0.8rem 2.4rem;
  margin: 3.2rem 0;
  border-radius: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  cursor: pointer;
}

.c0:hover {
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
}

.c0:disabled {
  background-color: #999999;
  color: #333333;
  cursor: not-allowed;
}

.c0:disabled:hover {
  -webkit-filter: none;
  filter: none;
}

.c0 > svg {
  width: 4rem;
  margin-left: 1rem;
}

<button
  class="c0"
  color="primary"
>
  Button
</button>
`;
